pkg/agents/team2/qtable.go

```diff
@@ -36,3 +40,33 @@ func (a *CustomAgent2) updateQTable(state int, action int) {
 	a.qTable[state][action] = (1-Alpha)*a.qTable[state][action] +
 		Alpha*(a.rTable[state][action]+Gamma*a.getMaxQ(a.CheckState()).maxQ)
 }
+
+func (a *CustomAgent2) exportQTable() {
+	f, err := os.OpenFile(fmt.Sprintf("%s%s%s", a.ID(), "qtable", ".csv"), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	if err != nil {
+		panic(err)
+	}
+	defer f.Close()
+
+	w := csv.NewWriter(f)
+	qTable := a.qTable
+	// make string table
+	sQTable := make([][]string, len(qTable))
+	for i := 0; i < len(qTable); i++ {
+		sQTable[i] = make([]string, len(qTable[0]))
+	}
+
+	// convert float64 qtable to string qtable
+	for i := 0; i < len(qTable); i++ {
+		for j := 0; j < len(qTable[0]); j++ {
+			sQTable[i][j] = fmt.Sprint(qTable[i][j])
+		}
+	}
+
+	//fmt.Println("rows:", len(sQTable), "columns:", len(sQTable[0]))
+	err = w.WriteAll(sQTable)
+
+	if err != nil {
+		log.Fatalln("error writing csv:", err)
```

Comment:
See above comment - does it have to be fatal?

Reply:
Fixed